Cult Improvement Drive

Xhuis opened the pull request:
This PR aims to make cult more engaging by changing rune functionalities, introducing new features, and changing objectives.

Changes:
- Cult structures now have a set amount of health and will break when destroyed
- Cult floors will now heal any cultists standing on them for 1 brute and 1 burn per tick.
- Tomes can now only be used when you are conscious, standing, etc.
- Cultists can no longer use null rods to erase runes
- Blind and Deafen talismans have been re-added
- Hide and Reveal Runes talismans now have 3 uses
- EMP runes are a new color (bluish-violet)
- The Corrupt rune has been added; upon use, it will cultify nearby turfs and transform any metal sheets into runed metal sheets

Commit:
Cult changes 1

Commit:
Cult improvement drive

Bawhoppen commented:
#ooh #ah

Anonus commented on code/game/gamemodes/cult/cult_structures.dm:
```diff
@@ -1,36 +1,139 @@
+// CULT TURFS //
+////////////////
+
+/turf/simulated/floor/plasteel/cult
```
You need to do this for `/turf/simulated/floor/engine/cult` as well, or preferably remove one of those paths.

KorPhaeron commented:
>Blind and Deafen talismans have been re-added

The reason this was removed is because a cultist being able to have a bag of 50 aoe "fuck you, you lose" papers was not fun. It won't be fun this time either.

Xhuis commented:
@KorPhaeron The only one that cripples nearby people is the Blind talisman, which only blinds for ten seconds.

KorPhaeron commented:
10 seconds of being unable to fight is gigantic. Thats twice the lenght of most stuns. It's aoe and entirely unblockable outside of a single item

And you can stack a ton of them in your bag so really its 100 seconds.

Trust me, we tried this, it was abused to hell, we removed it.

Metacide commented:
The rest sounds really neat though. Would be nice to see a dramatic fight over a properly-styled cult base.

Xhuis commented:
A lot of different runes are going to require certain structures. For instance, the Sacrifice rune now requires two pylons and an altar nearby.

Zilenan91 commented:
Xhuis, please, close this.

I'd offer criticism in any way I possibly could, but there isn't any. It's not that your changes are bad, but that cultists are such a mess of design, that by adding more and more nebulous "stuff" for them to do will just make the mode more unbearable than it already is. 

You put in base-building mechanics but don't remove their escape or mobility abilities. You add in blind and deafen talismans which were absolute cancer before and will be cancer now. You add in "cult weeds" to make cult autism forts even stronger and yet more unassailable due to the spamming of stun paper. But above all you do absolutely nothing about said stun paper, something that I'm honestly going to say doesn't just kill my enjoyment of cult mode, but kills my enjoyment of the entire game as a whole because of its sheer influence making me not want to play a majority of the jobs I'd enjoy.

Xhuis commented:
@Zilenan91 Due to your complete failure to read the fact that this is _work in progress_ and that _everything is subject to change_, I'm going to completely fail to read your post.

Zilenan91 commented:
@Xhuis If it's WiP, why isn't there a Work in Progress tag

KorPhaeron commented:
Xhuis can't add tags, only maintainers can

Zilenan91 commented:
@KorPhaeron Huh. I thought that creators of pull requests would be able to add their own tags for organization purposes. That's really bad design on githubs side.

duncathan commented:
@Zilenan91 that's what the big (WIP) in the PR title is for

Commit:
Lots more work

Xhuis commented:
New in this commit:
* By grabbing someone and clicking on a forge, you can DUNK THEIR FUCKING FACE INTO LAVA.
* Smithing! You can add an object to a forge and alt-click to attempt to modify it. For instance, turn a bedsheet into a cult version.
* cult_items.dm and cult_structures.dm have been moved into a new file under the cult folder, objects.dm.
* Cult walls can no longer be deconstructed with a welding tool
* Cult floors and walls can now be purified with a null rod
* Runed metal is now obtainable from the supply talisman in stacks of 20
* Many runes now require certain structures to work
* Boil Blood now deals 75 fire damage to non-cultists in addition to setting them on fire with 5 fire stacks
* Admin-only imbued tomes now exist for instantly drawing runes
* You can smith a bible to create an accursed tome, which draws runes one second faster
* Null rod wearers can no longer be harmed by tome-smacking

Anonus commented:
Cult walls should probably have some way for the cult to remove or move stuff through them.

Xhuis commented:
@Anonus That's a good point. I might make them removable via tomes.

Anonus commented on code/game/gamemodes/cult/runes.dm:
```diff
@@ -92,8 +94,10 @@ structure_check() searches for nearby cultist structures required for the invoca
+		user << "<span class='warning'>[src] lacks the nearby structures it requires!</span>"
 		return 0
 	if(!keyword && req_keyword)
+		user << "<span class='warning'>[src] requires no keyword!</span>"
```
requires a keyword, not no keyword.
though given that this won't normally appear, eh.

Anonus commented on code/game/turfs/simulated/floor/plasteel_floor.dm:
```diff
@@ -370,10 +370,6 @@
 /turf/simulated/floor/plasteel/vault
 	icon_state = "vault"
 
-/turf/simulated/floor/plasteel/cult
```
You need to remove lines 382 to 388, at least if that won't cause map errors.

Xhuis replied:
It will cause map errors, hence why I kept it in for legacy purposes.

Xhuis commented:
Cuboos has made a potential new Nar-Sie sound. Thoughts?
https://dl.dropboxusercontent.com/u/9654207/NarsieRisen.wav

Anonus commented:
Gotta convert nar-sie's tile replacement and the artificer spell to use `/turf/simulated/floor/engine/cult` but you probably know that.

Bawhoppen commented:
@Xhuis people are going to hate that sound, people will want to keep the old one
we saw how well the other sound went

Nitrousoxide commented:
>Cuboos has made a potential new Nar-Sie sound. Thoughts?
>https://dl.dropboxusercontent.com/u/9654207/NarsieRisen.wav

That sound is about 7 seconds too long.

Xhuis commented:
Nar-Sie spawns about five seconds into the sound.

Commit:
A lot more work

Xhuis commented:
New in this commit:
* The spell Lesser Construction now creates engine cult turf, as does Nar-Sie
* The tome menu now has a Cancel option
* Nar-Sie has a new sound
* The rune for summoning Nar-Sie now becomes distorted and shifting when the summoning begins (see here: http://i.imgur.com/Z6fA6Cm.gifv)

Robustin commented:
I dont care about blind/deafen, I've literally never seen them used and I'm one of the most experienced cultists OF ALL TIME, OF ALL TIME!

So I don't know why anyone's having a hissy fit over it when cult still has/had an AOE deathbomb and single target i-win talismans. Similarly, dont know why it should be re-added either, would just clutter up rune menu, maybe merge them into a blind+deaf rune (Call it Overawe or something pimp).

KorPhaeron commented:
>I've literally never seen them used

Take it from someone who has then. "You enter the same screen as a cultist and you lose" is not fun gameplay. There is literally no way to stop the blind outside of a single item. You can't fight at all. One cultist can take out the entire escape shuttle or all of security on their own because they will be blind for the whole fight. Blood boil requires a stationary rune and three people. Stun talisman requires you to melee them. Blind paper requires nothing.

We just nerfed AoE flashes for being the same nonsense and those actually had counters. 

I don't object to this update in general but it's frustrating to see you pushing ahead with the same mistakes (blind talisman, instant walls that are hard to get rid of) we already made, tested, and corrected after months/years of feedback on the game. Why do we have to go through the same trial and error again?

Xhuis commented:
By "instant walls", are you referring to wall runes? I can easily nerf those - perhaps make them return to normal after a minute or two.

Blind and Deafen talismans are staying, but may be modified drastically. I'm leaning towards making them melee and single-target with more powerful effects, similar to the Stun talisman.

KorPhaeron commented:
Single target would be fine though redundant, or making flash protection stop them (or do eyeblur only?) so they can be used on crowds of validhunters but not instantly cripple security

Zilenan91 commented:
@xhuis How about you make it so the unweldable cult walls require a corruption rune next to them to maintain their cult status? 
Activate the corruption rune to make them cult, slap another one on the floor to maintain their power.

Commit:
Conflicts 1/3

Commit:
Conflicts 2/3

Commit:
Conflicts 3/3

Xhuis commented:
New in this commit:
* New pylon breaking sound by Cuboos
* Fixed an unreported issue where the one-use talisman to summon tomes said it had two uses

Xhuis commented:
New in this commit:
* Cult floors and walls can now be deconstructed, yielding runed metal when destroyed.
* Hitting a cult floor or wall with a bible will weaken it and make it susceptible to deconstruction.
* Cultists cannot perform this interaction for obvious reasons as well as the ability to duplicate runed metal.

Xhuis commented:
This can be considered merge-ready once I push the changelog, but I want a few test rounds ran before this gets merged.

Aranclanos commented:
>This is DNM for now.

Xhuis commented:
@Aranclanos I forgot to update the OP. This isn't WIP anymore, nor is it DNM. I still want to run some test rounds before it gets in, though.

Xhuis commented:
New objective added: Bring the Slaughter. After invoking the new Call Forth The Slaughter rune, three cult-aligned, empowered slaughter demons will be summoned to kill the heretics still alive aboard the station. They also have the ability to locate any living beings that aren't cultists.

Xhuis commented on code/game/gamemodes/cult/cult.dm:
```diff
@@ -54,15 +54,16 @@
-	recommended_enemies = 4
+	required_players = 1 //20
+	required_enemies = 1 //4
+	recommended_enemies = 1 //4
```
oops debug code, one sec

Zilenan91 commented:
Can you make it so the demons have to kill the cultists too? Flavorful since nar-sie prioritizes cultists over regular crew.

Xhuis commented:
I removed the debug code. I also fixed an entirely unrelated issue where headslugs didn't have the right death icon.

@Zilenan91 I debated this but decided against it, as it might discourage the cultists from actually doing the objective until the last second, and it doesn't seem like it'd be very fun for the cultists who toiled to actually bring it about.

Zilenan91 commented:
@Xhuis but Nar-sie already kills everyone, I don't think the cultists are gonna give two shits if they die.

Xhuis commented:
Nar-Sie returns the cultists and everyone else as harvesters.

Zilenan91 commented:
In my experience being a harvester, while fun, is usually a complete waste of time. You just kinda run at dudes and GGNORE them with insta inject gas clouds of permastun.

Bawhoppen commented:
its end of round intentionally OP

Xhuis commented:
New in this commit:
* Stun talismans now operate off of a subtype of /talisman, /melee
* Blind and Deafen talismans are now melee talismans and have enhanced effects against single targets

Anonus commented on code/game/gamemodes/cult/cult.dm:
```diff
@@ -76,9 +77,12 @@
+	if(prob(33))
 		cult_objectives += "survive"
 		cult_objectives += "sacrifice"
+	if(prob(50))
```
Wouldn't this have a chance of giving a cult four objectives?

Xhuis replied:
Good point. I'll hotfix that.

Xhuis commented:
I just decided to straight-up rewrite how cult objectives are selected. It now chooses from a weighted list, with Escape having a probability of 20% - Nar-Sie and Slaughter both have a 40% chance.

Xhuis commented:
I'll squash commits in just a sec, but the minimum is 3 because of the merge conflict resolution.

Xhuis commented:
Commits squashed to as few as they'll go. If it's still a problem I can manually copy over everything to a new branch and have only one commit that way.

Xhuis commented:
This still needs earlymerge testing before it gets fully merged.

Commit:
Gives slaughter demons unique names in cult comms

Commit:
Optimizes some code

Xhuis commented:
New in these commits:
* Harbringers of the Slaughter now have a large name in cult communications
* Captains can now be converted to cult (but still must have their loyalty implant removed first)
* The convert rune now knocks out the target, instead of the conversion itself

KorPhaeron commented:
Wait what why are slaughter demons in cult

Xhuis commented:
People thought they were thematically fitting, and it seemed true enough to me. They're only summoned as a round-ending objective.

KorPhaeron commented:
Well I guess its not a balance issue if they're round end but I always imagined them unaffiliated with nar sie.

I'll put this in rotation/test merge it in a little while.

Dorsisdwarf commented:
I don't like that captains can be converted now.
I don't like converting the captain in any mode frankly

GunHog commented:
Why? No one should be *immune* to conversion except Chaplain, as he is protected by...god magic? If you can capture the Captain alive, you deserve to have that prize on your side, on par with managing to snag the HoS.

Dorsisdwarf commented:
Because the captain was always the one inconvertible guy, the one who always leads the defense, and if he was any good he was the last man standing. Now he'll only immune in 1/4 conversion modes, which is ew.

Hell, a few months ago the HOP was being considered for hardcoded cult immunity (His blandness of soul is too great even for the Geometer to find palatable)

GunHog commented:
HoP was removed from *starting* as cult. The reason you do not let HoP start as cult is because of all-access. Killing the HoP and taking his ID to spread access is not that much different than converting him for that same purpose.

Commit:
Fixes melee talismans being uncreatable

Xhuis commented:
Melee talismans, such as stun, are now creatable again.

Commit:
Makes the Reveal Runes talisman propery reveal them

adrix89 commented:
Captain has always had brainwashing protection training :-1:
Do you think Nanotrasen will have the captain be a liability?

Xhuis commented:
Yes.

Anonus commented:
Artificers can't use the floor or wall spells on centcom, which used to be needed due to them being able to crash the server but should no longer be needed now that the transit areas work differently.

Zilenan91 commented:
So it's all ogre now?

Anonus commented:
Well, if this is closed I guess I'll do my thing and give constructs cool action buttons and add a few cool effects.

Zilenan91 commented:
@Anonus It'd be best if you didn't, admins have been removing cult from rotation lately because it's just not good. Adding more things to it and buffing it won't solve the core problem of it being quite literally an abortion of design due to its creator leaving, and everything after that being tacked on.

Anonus commented:
I won't be buffing it, just slapping cool effects on it because effects are cool, and leaving constructs without action buttons is a tragedy.

duncathan commented:
@Zilenan91 did cult mode kill your parents or something

Zilenan91 commented:
@duncathan 
Yes

But for real, Cult is awful because at the core of it, it's a bunch of dudes who spawn with and can make a functionally infinite amount of parapens, gib people at will, summon unstunnable death bots, can 1-shot crit entire rooms at no effort, have a dedicated AI killer... I could go on. It has so many things about it that it just does not need and it feels very cheap.

duncathan commented:
@Zilenan91 personally I think that the core concept isn't flawed; it's the execution. It would take a lot of effort to truly do it well, though.

Zilenan91 commented:
Yeah, a total rework. That's why cult isn't able to be feasibly saved, too much work for anyone to ever really want to do. 

But yeah I meant core gameplay, not core concept, my bad.

Anonus commented:
I'm commenting here so I can link a screenshot to IRC because my internet is being really finicky.
![screenshot_18](https://cloud.githubusercontent.com/assets/3886584/11171544/cea2bb7c-8bc0-11e5-8903-2fc827acc426.png)

KorPhaeron commented:
Good stuff. What happened to this PR though? It's on server right now

Zilenan91 commented:
It's still under testing I think, never got merged.

Xhuis commented:
I'm no longer ridiculously frustrated.

Zilenan91 commented:
What in the fuck

Iamgoofball commented:
why did you quit xhuis i was looking forward to this
